src/middleware/attach-database-middleware.js

```diff
@@ -0,0 +1,11 @@
+const { getDatabase } = require("../utils/db");
+
+const attachDatabaseMiddleware = (request, response, next) => {
+  const db = getDatabase()
+  
+  response.locals.db = db
+
+  next()
+}
```

Comment:
This middleware injects the database instance into the `response.locals` object. Learn more about this pattern here: https://expressjs.com/en/api.html#res.locals